### PR TITLE
doc: add k8s landing page

### PR DESCRIPTION
**Pull request by eliecharra**
# Description of the change

This is to clarify the differences between multiple k8s operators we provide.

image

## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [ ] I have reviewed the [guidelines for contributing](https://github.com/spacelift-io/user-documentation/blob/main/CONTRIBUTING.md) to this repository.
- [ ] The preview looks fine.
- [ ] The tests pass.
- [ ] The commit history is clean and meaningful.
- [ ] The pull request is opened against the `main` branch.
- [ ] The pull request is no longer marked as a draft.
- [ ] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/user-documentation/blob/main/LICENSE) to Spacelift (not required for Spacelift employees).
- You have updated the navigation files correctly:
    - [ ] No new pages have been added, or;
    - [ ] Only _nav.yaml_ has been updated because the changes only apply to SaaS, or;
    - [ ] Only _nav.self-hosted.yaml_ has been updated because the changes only apply to Self-Hosted, or;
    - [ ] Both _nav.yaml_ and _nav.self-hosted.yaml_ have been updated.

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

_Spacelift employees should request reviews from the relevant engineers and are allowed to merge pull requests after they got at least one approval._

Thank you for your contribution! 🙇

### CODE REVIEW

File: docs/integrations/kubernetes/config-operator.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Spacelift Configuration Operator
```
Comment (eliecharra) [inline docs/integrations/kubernetes/config-operator.md:1]: No changes here, just a rename

File: nav.yaml [hunk 2]
```diff
@@ -151,6 +151,7 @@ nav:
           - integrations/source-control/raw-git.md
       - Kubernetes:
           - integrations/kubernetes/operator.md
+          - integrations/kubernetes/config-operator.md
```
Comment (adamconnelly) [inline nav.yaml:154]: Do we need to add this to the Self-Hosted nav as well? Just otherwise I think we'll end up with a broken link.

**Review by adamconnelly — COMMENTED** [review]
Other than one thing looks good.

File: nav.yaml (hunk 2 above)
Comment (eliecharra) [reply nav.yaml:154]: Ha yeah yeah good catch. I always forgot about the SH tree.

**Review by adamconnelly — APPROVED** [review]

**Pull request merged**